core/src/main/scala/io/finch/ContentTypeNegotiation.scala

```diff
@@ -0,0 +1,48 @@
+package io.finch
+
+import io.finch.internal.Accept
+import shapeless._
+
+trait ContentTypeNegotiation[ContentType] {
```

Comment:
How do you feel about naming this entity `NegotiateToResponse` so it takes an `Accept` and returns a `ToResponse` instance for it.